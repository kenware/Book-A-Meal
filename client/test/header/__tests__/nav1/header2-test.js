import React from 'react';
import { shallow } from 'enzyme';
import Header1 from '../../../../components/header/header2';

describe('Testing of sidebar component', () => {
  it('should render properly', () => {
    const tree = shallow(<Header1 />);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to click method', () => {
    const tree = shallow(<Header1 />);
    const wrapper = tree.instance();
    wrapper.onClick();
    tree.setState({ responsive: 'responsive' });
    wrapper.onClick();
    expect(tree).toMatchSnapshot();
  });

  it('should respond to logout method', () => {
    const tree = shallow(<Header1 />);
    global.history.pushState = jest.fn();
    const wrapper = tree.instance();
    wrapper.logout();
    expect(tree).toMatchSnapshot();
  });
});
