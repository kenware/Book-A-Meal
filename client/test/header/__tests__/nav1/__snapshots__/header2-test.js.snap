// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of sidebar component should render properly 1`] = `
<div>
  <div
    className="nav "
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "23px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span
      className="a"
      onClick={[Function]}
      role="button"
    >
      LogOut
    </span>
    <Link
      className="a"
      replace={false}
      to="/profile"
    >
      <em
        className="fa fa-user"
      />
        
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin"
    >
      Admin
    </Link>
    <Link
      className="a"
      onClick={[Function]}
      replace={false}
      to="/dashboard"
    >
      dashboard
    </Link>
    <Link
      className="a"
      onClick={[Function]}
      replace={false}
      to="/admin/allmeals"
    >
      All meals
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin/setmenu"
    >
      Set Menu
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin/addmeals"
    >
      Add meals
    </Link>
    <Link
      className="active a"
      replace={false}
      to="/meals"
    >
      Home
    </Link>
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;

exports[`Testing of sidebar component should respond to click method 1`] = `
<div>
  <div
    className="nav responsive"
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "23px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span
      className="a"
      onClick={[Function]}
      role="button"
    >
      LogOut
    </span>
    <Link
      className="a"
      replace={false}
      to="/profile"
    >
      <em
        className="fa fa-user"
      />
        
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin"
    >
      Admin
    </Link>
    <Link
      className="a"
      onClick={[Function]}
      replace={false}
      to="/dashboard"
    >
      dashboard
    </Link>
    <Link
      className="a"
      onClick={[Function]}
      replace={false}
      to="/admin/allmeals"
    >
      All meals
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin/setmenu"
    >
      Set Menu
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin/addmeals"
    >
      Add meals
    </Link>
    <Link
      className="active a"
      replace={false}
      to="/meals"
    >
      Home
    </Link>
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;

exports[`Testing of sidebar component should respond to logout method 1`] = `
<div>
  <div
    className="nav "
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "23px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span
      className="a"
      onClick={[Function]}
      role="button"
    >
      LogOut
    </span>
    <Link
      className="a"
      replace={false}
      to="/profile"
    >
      <em
        className="fa fa-user"
      />
        
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin"
    >
      Admin
    </Link>
    <Link
      className="a"
      onClick={[Function]}
      replace={false}
      to="/dashboard"
    >
      dashboard
    </Link>
    <Link
      className="a"
      onClick={[Function]}
      replace={false}
      to="/admin/allmeals"
    >
      All meals
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin/setmenu"
    >
      Set Menu
    </Link>
    <Link
      className="a"
      replace={false}
      to="/admin/addmeals"
    >
      Add meals
    </Link>
    <Link
      className="active a"
      replace={false}
      to="/meals"
    >
      Home
    </Link>
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;
